stl/inc/typeindex

```diff
@@ -40,9 +41,12 @@ public:
 
 #if _HAS_CXX20
     _NODISCARD strong_ordering operator<=>(const type_index& _Right) const noexcept {
-        return *_Tptr == *_Right._Tptr      ? strong_ordering::equal
-             : _Tptr->before(*_Right._Tptr) ? strong_ordering::less
-                                            : strong_ordering::greater;
+        // TRANSITION, DevCom-10326599, should rely on a stable interface
+        if (_Tptr == _Right._Tptr) {

```

Comment:
We should at least consider doing a:

```cxx
    return _CSTD __std_type_info_compare(reinterpret_cast<const __std_type_info_data*>(this), reinterpret_cast<const __std_type_info_data*>(&_Right)) <=> 0;
```

It very well might be better to duplicate the implementation here, but I'd like it considered. @stephantlavavej, opinion?

Reply:
@strega-nil-ms I asked the very same question - expand my comment above! :smile_cat: The answer is that `type_info` has a vptr, so we can't directly `reinterpret_cast`. Adding an offset to skip the vptr would be more squirrelly than the current approach, in my opinion.